apps/web/src/ui/components/pages/products/products-table/use-products-table-hook.ts

```diff
@@ -1,40 +1,211 @@
+import { useCallback, useMemo } from 'react'
+import { parseAsInteger, useQueryState } from 'nuqs'
 import { useApi, useCache } from '@/ui/hooks'
 import { ProductDto } from '@stocker/core/dtos'
+import { usePagination } from '@/ui/hooks/use-pagination'
 
-export const useProductsTableHook = (page: number) => {
-  const useapi = useApi()
+export const useProductsTableHook = () => {
+  const [pageState, setPage] = useQueryState('page', parseAsInteger)
+  const page = pageState ?? 1
+  const [filterByNameValueState, setFilterByNameValue] = useQueryState('name')
+  const filterByNameValue = filterByNameValueState ?? ''
+
+  const useApiInstance = useApi()
```

Comment:
Always desestructure the object returned by `useApi` hook to get its services inside it.